test/testSetup.ts

```diff
@@ -109,6 +124,11 @@ export class TestSetup {
   }
 
   public async createProject(scratchOrgEdition: string): Promise<void> {
+    if (this.usingExistingProject) {
+      utilities.log('Using existing project, skipping createProject()');
+      return;
+    }
+
```

Comment:
Having read through the PR, I think TestSetup is too muddy, and trying to do something it wasn't meant to do.

I think what would be better is:
1. revert the changes in testSetup.ts
2. in apexLspPerf.ts, in the first step, change it to:
```
  step('Ensure Apex Language Server Starts Successfully', async () => {
    utilities.log('ApexLsp - Set up the testing environment');
    testSetup = new TestSetup(
      'ApexLsp',
      false
    );

    await testSetup.setUpTestingEnvironment();
    .
    .
    .
```

In this suite, after `testSetup.setUpTestingEnvironment()` is called, the other functions in `setUp()` (`createProject()`, `authorizeDevHub()`, and `createDefaultScratchOrg()`) aren't used, so I think apexLspPerf.ts should manage its setup and tear down, and not `TestSetup`.

Reply:
IMO, the long game for me for TestSetup that will serve the test writer w/o having to decide if it can even be used or not.

I know this module is still under heavy development and the e2e test success is paramount, but I really would prefer not to do my own setup/teardown of what seems to me as a very common use case.